openhands/integrations/github/github_service.py

```diff
@@ -32,10 +33,55 @@ def __init__(
         external_auth_token: SecretStr | None = None,
         token: SecretStr | None = None,
         external_token_manager: bool = False,
+        base_url: str | None = None,
     ):
         self.user_id = user_id
         self.external_token_manager = external_token_manager
 
+        # Use config or environment variables for GitHub API URL
+        if base_url and base_url.strip():  # Check if base_url is not None and not empty
+            # For GitHub Enterprise Server, the REST API endpoint is /api/v3 and GraphQL is /api/graphql
+            # We just need to change the hostname
+
+            # Log the provided base_url for debugging
+            logger.debug(f'GitHub Service initialized with base_url: {base_url}')
+
+            # Remove any trailing slashes and whitespace
+            base_url = base_url.strip().rstrip('/')
+
+            # Check if this is already an API URL
+            if base_url == self.DEFAULT_BASE_URL:
+                # This is github.com API URL, use it as is
+                self.BASE_URL = base_url
+                self.GRAPHQL_URL = (
+                    get_github_graphql_url() or f'{self.BASE_URL}/graphql'
+                )
+            elif '/api/v3' in base_url:
+                # Already in the correct format for REST API
+                self.BASE_URL = base_url
+                # For GitHub Enterprise, GraphQL endpoint should be at /api/graphql
+                # But we need to maintain the full path structure
+                self.GRAPHQL_URL = f'{base_url}/graphql'
+            else:
+                # Assume this is a GitHub Enterprise Server hostname
+                # Append /api/v3 for REST API
+                self.BASE_URL = f'{base_url}/api/v3'
+                self.GRAPHQL_URL = f'{base_url}/api/graphql'
+
+            # Log that we're using GitHub Enterprise Server
+            logger.debug(
+                f'Using GitHub Enterprise Server with BASE_URL: {self.BASE_URL}'
+            )
+        else:
+            # Default to github.com API
+            self.BASE_URL = get_github_api_url() or self.DEFAULT_BASE_URL
```

Comment:
NIT: lets define the helper functions within this class (e.g `get_github_api_url`)

Let's remove the config class and just use user settings for storing the `BASE_URL` values